File: 240102/요일 맞추기/guess-day-of-week.cpp
#include <iostream>
#include <array>

using namespace std;

int main() {
    array<string, 7> str =  {"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};
    array<int, 13> day =  {0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};

    int m1, d1, m2, d2;
    cin >> m1 >> d1 >> m2 >> d2;

    int res1 = d1, res2 = d2;
    for (int i = 0; i < m1; i++) res1 += day[i];
    for (int i = 0; i < m2; i++) res2 += day[i];

    int idx = ((res2 - res1) % 7);
    if (idx < 0) idx += 7;

    cout << str[idx];

    return 0;
}

File: 250701/요일 맞추기/guess-day-of-week.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int m1 = sc.nextInt();
        int d1 = sc.nextInt();
        int m2 = sc.nextInt();
        int d2 = sc.nextInt();
        
        int[] days = new int[]{0, 31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31};
        String[] weekDays = new String[]{"Mon", "Tue", "Wed", "Thu", "Fri", "Sat", "Sun"};

        int start = d1;
        for (int i = 0; i < m1; i++) {
            start += days[i];
        }
        int end = d2;
        for (int i = 0; i < m2; i++) {
            end += days[i];
        }

        int min = end - start;
        
        String res = weekDays[(min % 7 + 7) % 7];
        System.out.println(res);
    }
}